demo/api/schema.gql

```diff
@@ -614,6 +615,7 @@ type Query {
   userPermissionsAvailablePermissions: [String!]!
   userPermissionsContentScopes(userId: String!, skipManual: Boolean): [JSONObject!]!
   userPermissionsAvailableContentScopes: [JSONObject!]!
+  currentUserContentScopes: [JSONObject!]!
```

Comment:
what is the difference between currentUser.contentScopes and this?

Reply:
Why not put it unter currentUser into a FieldResolver?

And I don't understand how this improves performance, especially as you always load currentUserContentScopes in CurrentUser query.

Reply:
> The main speed improvement is in creating the currentUser-Object (which is done at every API-request).

I understand that. And I have the feeling that creating the full currentUser-Object every time (vs. lazy loading required data) is not good for performance and should possibly changed. But maybe you addressed the only bottleneck already with this PR.

> So I decided for a separate resolver

:+1: that makes it lazy loaded

> furthermore a field resolver requires an explicit ObjectType which we don't have for ContentScopes

it has to be a field resolver for CurrentUser, why shouldn't that be possible?

Reply:
The `UserContentScopesResolver` needs a `@Resolver(() => CurrentUser)`, as CurrentUser is the parent. (just like the error message says it)

In your case it is difficult though, as currentUser is returned by AuthResolver which gets the currentUser class injected. What is the use-case for extending CurrentUser? Is that still needed now that application specific settings as eg. domains are now implemented using contentScopes?